src/ansys/fluent/core/codegen/walk_api.py

```diff
@@ -0,0 +1,48 @@
+"""Module containing tool for walking (generated) API class hierarchy.
+
+Example
+-------
+
+.. code-block:: python
+
+    >>> from ansys.fluent.core.codegen import walk_api
+    >>> from ansys.fluent.core.generated.solver import settings_252
+    >>> walk_api.walk_api(settings_252.root, lambda p: print(p), current_path=[])
+
+"""
+
+from typing import List
+
+
+def walk_api(api_root_cls, on_each_path, current_path: str | List[str] = ""):
+    """
+    Recursively traverse the API hierarchy, calling `on_each_path` for each item.
+
+    Parameters:
+    - api_root_cls: The root class of the API hierarchy.
+    - on_each_path: A callback function to call for each path.
+    - current_path: The current path in the hierarchy (default: empty string).
+      Paths can be either dot-separated strings or string lists. The type is
+      determined by the client.
+    """
+    # Skip the root path
+    if current_path:
+        on_each_path(current_path)
+
+    # Get child names and their respective classes
+    child_names = getattr(api_root_cls, "child_names", [])
+    child_classes = getattr(api_root_cls, "_child_classes", {})
+
+    # Traverse each child
+    for child_name in child_names:
```

Comment:
We also need to loop over `command_names` and `query_names`

Reply:
and `argument_names` for Command and Query classes

Reply:
In a later version, I also want to pass each item's base type info to the callback for type-specific checks.